docs: reorganize documentation

petertonysmith94 opened the pull request:
Closes #1899

Re-organize the menu and page structure for the Docs

---

### Summary

- Seperated our docs into the following sections `Introduction`, `Basics`, `Essentials`, `Extras`, `Tooling`.

Before             |  After
:-------------------------:|:-------------------------:
<img width="254" alt="Screenshot 2024-03-21 at 13 47 45" src="https://github.com/FuelLabs/fuels-ts/assets/16990131/93a2a01f-beab-485c-963d-d9b4ed5184f7"> |  <img width="254" alt="Screenshot 2024-03-21 at 13 47 45" src="https://github.com/FuelLabs/fuels-ts/assets/16990131/53085730-3e6a-4f3c-95c3-d58e574b2729">

petertonysmith94 commented on apps/docs/src/extras/utilities/address-conversion.md:
```diff
@@ -1,4 +1,24 @@
-# Converting Between Address Formats and Native Types
+# Address
```
Concatenated `guide/address/conversion.md` and `guide/address/index.md` together to get the `address-conversion`

petertonysmith94 commented:
Pipeline currently [failing](https://github.com/FuelLabs/fuels-ts/actions/runs/8365980048/job/22905114913?pr=1906#step:6:1174) on an assertion for the expected structure.

We could update the action [FuelLabs/github-actions](https://github.com/FuelLabs/github-actions/blob/master/docs-hub/vp-docs.js#L15C3-L15C46).

danielbate commented:
Nice work, I think my preference is option B as I'm concerned about compatibility with the docs hub.

petertonysmith94 commented on apps/docs/src/guide/wallet-manager/index.md:
Consolidated to `apps/docs/src/guide/wallets/wallet-manager.md`

petertonysmith94 commented on apps/docs/src/guide/wallet-manager/locking-and-unlocking-wallet-manager.md:
Consolidated to `apps/docs/src/guide/wallets/wallet-manager.md`

petertonysmith94 commented on apps/docs/src/guide/wallets/wallet-manager.md:
```diff
@@ -1,3 +1,21 @@
+
+The `WalletManager` is a robust tool designed for managing vaults of wallets. It offers robust management of vaults, including support for custom storage and powerful encryption of all held vaults.
+
+## Key Features
+
+### Managing Vaults with `WalletManager`
+
+This includes adding new wallets to specific vaults, retrieving all wallets from a vault, exporting specific vaults, and exporting private keys. The `WalletManager` class currently supports two types of vaults: `PrivateKeyVault` and `MnemonicVault`.
+
+### Custom Storage Solutions
+
+The `WalletManager` supports defining a custom storage solution, allowing you to specify how and where the encrypted vaults are saved. With support for custom storage, you can make the `WalletManager` to fit your specific needs and security requirements.
+
+### Locking and Unlocking `WalletManager`
+
+The `WalletManager` implements an automatic encryption mechanism, securely saving the wallet's held vaults. This not only preserves the state of your vaults but also ensures robust protection of the stored information. When needed, you can easily unlock and decrypt the vaults using the previously defined password.
```
This can be simplified IMO

petertonysmith94 replied:
Will leave for https://github.com/FuelLabs/fuels-ts/issues/1919.

petertonysmith94 commented on apps/docs/src/guide/wallets/wallet-manager.md:
```diff
@@ -1,3 +1,21 @@
+
+The `WalletManager` implements an automatic encryption mechanism, securely saving the wallet's held vaults. This not only preserves the state of your vaults but also ensures robust protection of the stored information. When needed, you can easily unlock and decrypt the vaults using the previously defined password.
+
 # Getting Started with `WalletManager`
```
Wrong heading indent

petertonysmith94 replied:
https://github.com/FuelLabs/fuels-ts/commit/2d63f05a58db07b40771289abc9626fbb92f7ccc

petertonysmith94 commented on apps/docs/src/guide/providers/querying-the-chain.md:
```diff
@@ -32,3 +32,23 @@ Get all the spendable balances of all assets for an address. This is different f
 This method returns all the blocks from the blockchain that match the given query. The below code snippet shows how to get the last 10 blocks.
 
 <<< @/../../../packages/account/src/providers/provider.test.ts#Provider-get-blocks{ts:line-numbers}
+
+### Get messages
+
+You can use the `getMessages` method to retrieve a list of messages from the blockchain.
+
+<<< @/../../../packages/fuel-gauge/src/coverage-contract.test.ts#Message-getMessages{ts:line-numbers}
+
+### Get resources
+
+You can use the `getResourcesToSpend` method to retrieve a list of all the resources (coins + assets) that can be spent by a given address.
+
+<<< @/../../../packages/account/src/account.test.ts#Message-getResourcesToSpend{ts:line-numbers}
+
+### Get message proof
+
+A message proof is a cryptographic proof that a message was included in a block. You can use the `getMessageProof` method to retrieve a message proof for a given transaction ID and message ID.
+
+<<< @/../../../packages/account/src/account.test.ts#Message-getMessageProof{ts:line-numbers}
+
+<!-- TODO: Add docs for the two new parameters `commitBlockId` and `commitBlockHeight` -->
```
Consolidated various message docs into here

petertonysmith94 commented on apps/docs/src/guide/predicates/predicate-with-more-complex-args.md:
Moved to `apps/docs/src/guide/predicates/instantiating-a-predicate.md`

petertonysmith94 commented on apps/docs/src/guide/predicates/index.md:
```diff
@@ -17,24 +17,6 @@ The key difference here is that instead of checking these rules directly on the
 
 This method is not only more efficient but also helps to prevent traffic jams on the network and makes transactions cheaper. It does so by reducing the need for repetitive calculations on the blockchain.
 
-## Instantiating predicates
-
-A predicate in Sway can be as simple as the following:
-
-<<< @/../../docs-snippets/test/fixtures/forc-projects/return-true-predicate/src/main.sw#predicate-index-1{rust:line-numbers}
-
-In this minimal example, the `main` function does not accept any parameters and simply returns true.
-
-Just like contracts in Sway, once you've created a predicate, you can compile it using `forc build`. For more information on working with Sway, refer to the <a :href="introUrl" target="_blank" rel="noreferrer">Sway documentation</a>.
-
-After compiling, you will obtain the binary of the predicate and its JSON ABI (Application Binary Interface). Using these, you can instantiate a predicate in TypeScript as shown in the code snippet below:
-
-<<< @/../../docs-snippets/src/guide/predicates/index.test.ts#predicate-index-2{ts:line-numbers}
-
-The created [`Predicate`](../../api/Account/Predicate.md) instance, among other things, has three important properties: the predicate `bytes` (byte code), the `chainId`, and the predicate `address`.
-
-This address, generated from the byte code, corresponds to the Pay-to-Script-Hash (P2SH) address used in Bitcoin.
-
```
Moved to `apps/docs/src/guide/predicates/instantiating-a-predicate.md`

petertonysmith94 commented on apps/demo-fuels/fuels.config.ts:
```diff
@@ -9,5 +9,5 @@ export default createConfig({
 /**
  * Check the docs:
- * https://fuellabs.github.io/fuels-ts/guide/cli/config-file
+ * https://fuellabs.github.io/fuels-ts/tooling/cli/fuels/config-file
```
```suggestion
 * https://fuellabs.github.io/fuels-ts/guide/fuels/config-file
```

petertonysmith94 replied:
https://github.com/FuelLabs/fuels-ts/commit/9a9dbbb9938414dc564426bc19383efde8550e00

petertonysmith94 commented on apps/docs/src/guide/cli/index.md:
````diff
@@ -1,53 +1,7 @@
-```sh
-my-fuel-dapp # NextJS app or similar
-├── sway-programs # Forc's workspace
-│   ├── src
-│   ├── ...
-│   └── Forc.toml
-├── public
-│   └── ...
-├── src
-│   ├── app
-│   ├── ...
-├   └── sway-programs-api # Type-safe generated API
-└── package.json
-```
-
-## Installation
-
-Add it to your `my-fuel-dapp` project:
-
-::: code-group
-
-```console [npm]
-npm install fuels --save
-```
-
-```console [pnpm]
-pnpm add fuels
-```
-
-:::
-
-## Double-checking
-
-```console
-npx fuels -v
-```
-
-## Next Step
-
-Use [`pnpm fuels init`](./commands#init) to create a [`fuel.config.ts`](./config-file) file.
````
Moved to `apps/docs/src/guide/fuels/index.md`

petertonysmith94 commented on apps/docs/src/guide/contracts/index.md:
```diff
@@ -1,14 +1,7 @@
-# Contracts
+# Deploying Contracts
```
```suggestion
# Contracts
```

petertonysmith94 replied:
https://github.com/FuelLabs/fuels-ts/commit/4b47091c942ac4855b9bccc496f0fe220b3a5828

danielbate reviewed (approved):
Great work, good to get this in before we start to evolve the pages.

Approving as this does meet the specification for the restructure.

Couple thoughts:
- Are we keeping glossary, it's not in the new spec? We don't really maintain it and there and the docs hub has a glossary.
- Not sure how I feel about the contents tables, think I preferred the blurb about each section. And also it means we have to maintain 2 orders per section (links json and contents page)

arboleya reviewed (changes requested):
Two things only:
 1. Initialize all main sections expanded
 3. Remove the Table of Contents page from all of them
    - Not only because it will be troublesome to maintain, but especially because it will not be present in the docs hub.

**Edit**: And yes, the glossary can go as well.

petertonysmith94 commented:
@arboleya and @danielbate - thanks for reviewing 🙏🏼 

Made the following changed, based on your comments:

- Removed table of contents
https://github.com/FuelLabs/fuels-ts/pull/1906/commits/a9eba896166a597199ef1b78d611f84dd6eeafcf
- Removed glossary
https://github.com/FuelLabs/fuels-ts/pull/1906/commits/653f7f04826cfbb03d39b0fe60fd18b5b26df9bf
- All main sections expanded
https://github.com/FuelLabs/fuels-ts/pull/1906/commits/b1588540ccf8117eccd5d0b88b24750494cdeb82
